File: gripl/gripl-frontend/src/models/dto/DefaultPromptSelectionOverview.ts
/**
 * Compact data required to populate the default prompt selector.
 *
 * It contains the selectable versions for every prompt and the currently
 * configured default selection. The default selection is null when no prompt
 * version has been configured as the default.
 */
export interface DefaultPromptSelectionOverview {
    prompts: PromptSelectionOption[];
    defaultSelection: DefaultPromptSelection | null;
}

/**
 * Represents the available prompt-version choices for one prompt in the
 * default prompt selector.
 */
export interface PromptSelectionOption {
    promptId: number;
    versions: PromptVersionSelectionOption[];
}

/**
 * Represents lightweight metadata for a prompt version selectable as default.
 */
export interface PromptVersionSelectionOption {
    id: number;
    versionNumber: number;
}

/**
 * Identifies the prompt and prompt version configured as the current default.
 */
export interface DefaultPromptSelection {
    promptId: number;
    promptVersionId: number;
}

File: gripl/gripl-backend/src/main/kotlin/de/mertendieckmann/griplbackend/model/dto/DefaultPromptSelectionOverview.kt
package de.mertendieckmann.griplbackend.model.dto

/**
 * Lists every prompt that has at least one version together with lightweight 
 * version metadata. The default selection is null when no prompt version is 
 * configured as the default for regular analyses.
 */
data class DefaultPromptSelectionOverview(
    val prompts: List<PromptSelectionOption>,
    val defaultSelection: DefaultPromptSelection?
)

/**
 * Represents the selectable prompt versions belonging to a prompt.
 */
data class PromptSelectionOption(
    val promptId: Long,
    val versions: List<PromptVersionSelectionOption>
)

/**
 * Represents lightweight metadata for a prompt version selectable as default.
 */
data class PromptVersionSelectionOption(
    val id: Long,
    val versionNumber: Int
)

/**
 * Identifies the prompt and prompt version configured as the current default.
 */
data class DefaultPromptSelection(
    val promptId: Long,
    val promptVersionId: Long
)

/**
 * Metadata of a prompt version used for building the DefaultPromptSelectionOverview.
 */
data class DefaultPromptSelectionCandidate(
    val promptId: Long,
    val promptVersionId: Long,
    val versionNumber: Int,
    val isDefault: Boolean
)